pipeline/fudge_ci.py

```diff
@@ -0,0 +1,282 @@
+        """make data consistent for a given bound"""
+        higher_arr = higher_ds[cbound].values
+        lower_arr = lower_ds[cbound].values
+
+        # mask represents the indices where the higher duration is less
+        # than the lower duration
+        diff = higher_arr - lower_arr
+        mask = diff <= 0
+        num = np.count_nonzero(mask)
+
+        # Adjust values
+        if num > 0:
+            print(f"    {num} value(s) changed", flush=True)
+            values_changed += num
+            higher_arr[mask] = lower_arr[mask] * 1.001
+        
+        return higher_arr, values_changed
+
+    print(" Iterating over durations...", flush=True)
+
+    values_changed = 0
+
+    # Iterate over pairs of durations (higher and lower)
+    for (i, higher_dur) in enumerate(DURATIONS[1:], start=1):
+        lower_dur = DURATIONS[i - 1]
+        print(f" {lower_dur} -> {higher_dur}", flush=True)
+
+        # Naturally, we want to load files from the input directory and save
+        # them into the output directory, but since each file is likely
+        # to be loaded/saved multiple times as we change values, we need to be
+        # sure to only load from the input directory the first time and from the
+        # output directory all other times (so that changes to values persist
+        # as we do multiple iterations). So we track which durations we've
+        # "visited" before and choose the directory to load from accordingly.
+        if visited_durations[i]:
+            src_dir = out_path
+        else:
+            src_dir = path
+            visited_durations[i] = True
+            visited_durations[i - 1] = True
+
+        higher_files = glob(
+            os.path.join(src_dir, f"*_{data_group}_sum_wrf_{higher_dur}_*.nc")
+        )
+
+        for higher_file in higher_files:
+            # Durations are stored across files, so we need to get the corresponding
+            # files for the higher and lower durations
+            lower_file = higher_file.replace(higher_dur, lower_dur)
+
+            # Load data
+            higher_ds = xr.load_dataset(higher_file)
+            lower_ds = xr.load_dataset(lower_file)
+            #             higher_arr = higher_ds['pf'].values
+            #             lower_arr  = lower_ds['pf'].values
+
+            #             # mask represents the indices where the higher duration is less
+            #             # than the lower duration
+            #             diff = higher_arr - lower_arr
+            #             mask = diff <= 0
+            #             num = np.count_nonzero(mask)
+
+            #             # Adjust values
+            #             if num > 0:
+            #                 print(f"    {num} value(s) changed", flush=True)
+            #                 values_changed += num
+            #                 higher_arr[mask] = lower_arr[mask] * 1.01
+            #                 higher_ds['pf'][...,...,...] = higher_arr
```

Comment:
Should this be deleted or is it commenting on the structure of an algorithm?

Reply:
Oh thanks, nope that stuff was moved into the `fudge_cbound` function since it has to be done twice in this case (upper and lower). Will fix.